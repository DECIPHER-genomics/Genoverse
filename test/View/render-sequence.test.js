const { afterTest, testTrackRenderStatic } = require('../utils');

describe('Correctly render sequence:', () => {
  afterEach(afterTest);

  const track = {
    model     : Genoverse.Track.Model.Sequence,
    view      : Genoverse.Track.View.Sequence,
    resizable : 'auto',
  };

  describe('ATCGN', () => {
    const sequence = 'ATCGN';

    it('without labels',         () => testTrackRenderStatic(sequence, track));
    it('with labels',            () => testTrackRenderStatic(sequence, track, { start: 1, end: 5 }));
    it('with lower case labels', () => testTrackRenderStatic(sequence.toLowerCase(), { lowerCase: true, ...track }, { start: 1, end: 5 }));
  });
});

describe('Correctly render sequence variation:', () => {
  afterEach(afterTest);

  const track = {
    model     : Genoverse.Track.Model.SequenceVariation.extend({ seqModel: Genoverse.Track.Model.Sequence.extend({ data: 'ATCGN' }) }),
    view      : Genoverse.Track.View.SequenceVariation,
    resizable : 'auto',
  };

  describe('T/C', () => {
    it('without labels', () => testTrackRenderStatic([{ start: 2, end: 2, ref_allele: 'T', alt_allele: 'C' }], track));
    it('with labels',    () => testTrackRenderStatic([{ start: 2, end: 2, ref_allele: 'T', alt_allele: 'C' }], track, { start: 1, end: 5 }));
  });

  describe('TC/-', () => {
    it('without labels', () => testTrackRenderStatic([{ start: 2, end: 3, ref_allele: 'TC', alt_allele: '-' }], track));
    it('with labels',    () => testTrackRenderStatic([{ start: 2, end: 3, ref_allele: 'TC', alt_allele: '-' }], track, { start: 1, end: 5 }));
  });

  describe('T/AGA', () => {
    it('without labels', () => testTrackRenderStatic([{ start: 2, end: 2, ref_allele: 'T', alt_allele: 'AGA' }], track));
    it('with labels',    () => testTrackRenderStatic([{ start: 2, end: 3, ref_allele: 'T', alt_allele: 'AGA' }], track, { start: 1, end: 5 }));
  });

  describe('bumped variants', () => {
    it('without labels', () => testTrackRenderStatic([{ start: 2, end: 3, ref_allele: 'TC', alt_allele: 'AGA' }, { start: 3, end: 4, ref_allele: 'CG', alt_allele: 'A' }], track));
    it('with labels',    () => testTrackRenderStatic([{ start: 2, end: 3, ref_allele: 'TC', alt_allele: 'AGA' }, { start: 3, end: 4, ref_allele: 'CG', alt_allele: 'A' }], track, { start: 1, end: 5 }));
  });
});
