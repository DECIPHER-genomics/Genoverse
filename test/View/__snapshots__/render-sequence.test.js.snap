// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render sequence variation: T/AGA with labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAAFYklEQVRoge2aS2hUVxzGf2cmM3m/rImR+EjM+IoNCQkGEhQVxU2w0IWixYWJSSi40hRXBemi4CK6UKSgBCEKLZaidiO0KEqwKVONwXHqVMakEvNOzWQeeY2Z08Uko7F53jmTjPT+NjP/c8/95oP/nHvnnm9EXW+dRBE3+s4hlakFEQIqfr2vTO+7k3eUaQE8fnxQqV6MUrVJKu8WKdG5urdVic7HhGG5DehMZ0ErxNPl4flPzymuKSYmTuOiEoK4TzfPeGjsrzbk+Lg2XSAjLZ5Na1KJNcfQ3uWmvcetWcvpHKSp6TWDg6Ns3foJ+/fnYjQKTVqBQIBr166xfft28vPzAbDb7fj9foqKZr6KLGiF2H+w01zfTNsvbZqMhQwOjxAYHsGYkkTMyhWhOpwbT9m2LE4eLCQlMRak5EB5DuXbsjRpNTQ8Zffu7+nocBMba6Sx8RmHD9/W7E1KyaVLlzh79mxorKWlBavVOus5C/q6O2452Pn1Thw3HWz6bJNWd4y/fAWAiInBkBAfqrWSFG/i0G4L315/xMDQKAD3nnSSFG9atFZf3zCnTt2jtbWS3NxUAE6cKOblS1dYHhMTE0lOTqa5uZmysrJ558+7Qvrt/ZgSTBQdK6L7STdjQ2NhGVRJ7uoU/u5xh5oxhXfEv2gtq7Wb0tLVoWZMkZeXFpZHgOPHj3PlypUFzZ23IY6bDrZ8vgWDycCGfRtw3nGGbVAVaUlm3D7t95736ez0kJWVGKrv3n3F5cutNDY+C1u7uLgYo9FIS0vLvHPnbogMXq7cHW6sF634fX4ctxxhG1SFyztOSqJZiVZ2djI9Pb5pY2NjE9TV3VOiX1lZSUNDw7zz5mxI16MuTAkmElcFvzkr81cy8HyA4YFhJSbDpb3bTU5WChlp8dPGkxMWfw8pLV2N1dpNe/sQAHv3rqeyskCJT4Dy8nI8Hg82m23OeXPe1B03HRQcLaCktiQ05u5w8+LnFxRVqXn4CwfviJ8f7zupO1hEk62LMf8E+etXYHX08vufvYvSysxMoL5+Dzt2XKe6upDs7CSePu3HYklX5reqqorTp09jsVhmnTNnQ1LXpbL5wPRnh8JjhfTb+8My9nZgEGHyhKUxxW/2HpydQ2xck0asycDth+286tWmXVNTyJ4962hqeo3LNUpFRR4XLuzT7E0IwZEjR0L1rl27qK6unvUZBOZpSMmXJf8Zy8jPICM/Q7NJgIl/3oR1/of0uUboc40o0bJY0pWtCoPBQG1tbagWQkyrZzxHySfrKENvSJShbZNmFtbakFICkm+UCArOCAEdBWp9RjMR2X5XzVobbwAX4AO8gFdKXGKylga8hgCugMAnJF4BXkTw+AR4DRJvwIgrzozPuZHo2WqYgWhvSEBKDALSEaRLObmkRTC4ksG3CAlSBF+ZHId3xwMCRADGRmGtDT/BpkZlg6P7kgVIgVEEMCOIk2AWAjMSk5isATMSs4Q4ISZrMAFxEKrN79WGqaa+32B418B3H/7hwIzD/o4C1GwXEIEVIgQcWlV3RpVefaZai1/1vVWmdaPvHASbrwz9V1aUoTckyliWhni6PFgvWnk7Gv7lw+kc5OpVG+fP/8GdO21MTIT/txeV/hbLsjREVSSsOnJV7U8Ly/KzV0UkHKnIVZU/rSz5ClEVCUcqcl3uyHrJG6IqEo5U5LrckfXSNkRhJByRyDUKIuslbYjKSDgSkWs0RNZLelNXGQlHInKNhsh6SVfIbJGwOVnbVlBNTSEPHnxBTk4qPp+fioo8Hj48GjX+tKB8cxHgUGadMs2PYC9LaV6jb51EGRFZIf839BWio6Ojo6OjoxPt/AsofYxFLfKCiwAAAABJRU5ErkJggg=="`;

exports[`Correctly render sequence variation: T/AGA without labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAABTElEQVRoge3Yv0qCURzG8e+vP4tLU4LCewOB0OAdtAjNtXcJYXu059zo7g04RNANvBF5ARFOiXNLf06DEL68CpJaD/V8Jt/j4XDgy9GXE+3ndmJKb9QhFUYmIuDw+rYwdnXaL83L86PSWLPZjPKKNsvWvC9Obva/PncP7n9kMwYbv70BK3IQMQ4ixkHEOIgYBxHjIGIcRIyDiHEQMQ4ixkHEOIiY0rV4NiClBCQupmadR8CwUZ5vqzX3+n2W7IE+wV2CfHuT/HGPp3Vt7L9aNMhLgkoELaAVwOs7ZAPGQA7kjrQai/1kARHsfEA9oAbUgRpBZXqBNFlwPGywu9Zd/2EzgwAcV9uF8ctq+TCdjd4Kz71RB/B/zTL8liXGQcQ4iBgHEeMgYhxEjIOIcRAxDiLGQcQ4iBgHEeMgYube9i7Dt73f5xNiZmZmtg6fsEFA07qbdmMAAAAASUVORK5CYII="`;

exports[`Correctly render sequence variation: T/C with labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAAEYUlEQVRoge2aXUhbZxzGfycxWo1f7Uyn2LW1ZrZNJwZFQWlRseuNtLALR4VdNH4x6FUreDUouxh44W46ZFCRgRY2NobtYBS2KYJsjtBa0WZmJerE+u2mJmqMmZ5dOMPc1Og579I0nN/d++Y9z3ngOf/z8p5/pPqZehlBfDn7MbIwtS0kCcq/7xam9+mtR8K0AJ48qRCqFyVU7W9snVYhOp+V9QvReZXQvWwDGjs5UIV4Jj0MfT1Ebm0uUUcUFpUkceSts7v+5Pt1BHl9XZkuYEqOJetEEjHRUYxOuhmddivWcrkW6Ol5wcLCGufPv8aVKxno9ZIirc3NTdrb28nPz8disQDgcDjw+/1Yrbu/RQ5UIY4vHPQ29TLy3YgiYwGDq142V73oE+OJSjkWGKvZeAovpHKrIodEYwzIMleLTlN0IVWRVmvrACUlnzM+7iYmRk9b2zOuX3+o2JssyzQ3N9PY2BiY6+vrw26373nNgR535wMnlz64hLPDSda1LKXuWB8eA0CKikIXFxsYKyU+1sC7JWY+uv+Y+aU1ALqeThAfazi01uzsKrdvd9HfbyMjIwmAmzdzGR5eVOXRaDSSkJBAb28vhYWFQdcHrZA5xxyGOAPWG1amnk7hW/KpMiiSjLREfpt2B8LYZtnrP7SW3T5FQUFaIIxtMjOTVXkEqK6upqWl5UBrgwbi7HBy7p1z6Aw6zlw+g+uRS7VBUSTHR+NeUb73/JOJCQ+pqcbAuLNzjHv3+mlre6ZaOzc3F71eT19fX9C1+wcib72u3ONu7J/Y8a/4cT5wqjYoisXldRKN0UK00tMTmJ5e2THn821QX98lRN9ms9Ha2hp03b6BTD6exBBnwPj61pOTYklhfmie1flVISbVMjrl5nRqIqbk2B3zCXGH30MKCtKw26cYHV0CoKzsFDZbthCfAEVFRXg8HgYHB/ddt++m7uxwkv1eNnl1eYE597ib5988x1ol5vCnhmWvn6+6XdRXWOkZnMTn38By6hh25ww//zJzKK3jx+Noairl4sX71NTkkJ4ez8DAHGbzUWF+q6qqaGhowGw277lm30CSTiZx9urOs0POjRzmHHOqjP05v4Bk8KjS2OYnxzSuiSXePJFMjEHHwx9HGZtRpl1bm0Np6Ul6el6wuLhGeXkmd+9eVuxNkiQqKysD4+LiYmpqavY8g0CQQPLez/vPnMliwmQxKTYJsPH7H6qu/zezi15mF71CtMzmo8KqQqfTUVdXFxhLkrRjvOs1Qu6sIQwtkDBD2UeaPXhjEFmWAZkPhQhK3JEkGM8W6zOc0SokzNACCTO0QMIM4R1DSYLyH7rviND69u0SETKvFFqFhBlaIGHG//Inh2CIbLlGGiGvEJEt10gkpBUisuUaqYS0QkS2XCOVkAYisuUaqYQ0EJEt10glpIGIbLlGKiHd1EW2XCOVkJ9DRLZcI5GXcjAU2XKNNLRPJ2GG8I6hSL1ttI6hhoaGhoaGhoaGOv4CpfGPY9uABK4AAAAASUVORK5CYII="`;

exports[`Correctly render sequence variation: T/C without labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAAA80lEQVRoge3ZsQnCQBSH8ffENRxBsMgAglhZO4AjSAYIDmBtaW8vFipkgYDgCFa6xFmIknBJZ+If/H6VOR/h4OOanKf3NFjJ7rG2UFl5cTebHfPK2mZ5iOaKYh6tJUni8RtRp9/0x+I8+vzeTi6dbAZmvV9vAFUEEUMQMQQRQxAxBBFDEDEEEUMQMQQRQxAxBBFDEDHRZ/HB1UIIZhZsVZrK3M1uw3ge38UJEUMQMQQRU3tj6G42O+XZ+3k/HXe3oz/HCRFDEDEEEUMQMQQRQxAxBBFDEDEEEUMQMQQRQxAxBBFTe2PYNMyNYfs4IQAAAG14AhmIHcpE0vW0AAAAAElFTkSuQmCC"`;

exports[`Correctly render sequence variation: TC/- with labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAAEvElEQVRoge2ZXUxTZxzGf+/pB0hLwY86DHMTbRS7EggICUSji4s3ZiNeLJHFC0EkS7yThCXLkmV3XrirxRsNyYJLtpsFt5slSzQxxjRplJkC2hnQYZHvCVJaaUt7dlHphgNaTl/wuJ3fzel7+r7Pec55zv+850O0j7erSOL0ga9RpamtD0KA58ln0vTu3v1YmhaAWaraS6wlVdK0bpfJ06r1fitNa71QXrcBg6VkVSGhkRAPfnxA9dlqzPkai0oI8j37lv0r+vsj1FhMmy5QkG9mqyMfs0lhOhRlZi6qWWtgYJpbt4aZnp5n//6tHDtWhskkNGklk0muXr1KbW0tbrcbgP7+fuLxOFVVy1d+VhXS/0M/3oteHv36SJOxtMHIC5KRF5gcdszbtqTbuUw8bzvt1LtLyLOYQFXZt7OYnU67Jq3OTj9HjnxPMDhLXp6Jrq4+Tp78SbM3VVW5dOkSFy5cSK/r6enB5/OtOCar0z1wLcChLw4R6A6w96O9Wt0RGxwCQJjNKAWb0m2tWM0K7+3awi3/CJHoAgCPx0JYzWu/Ek9MRDh//gb37jVTVlYEwLlz1QwOzuTk0WazUVhYiNfrpb6+PmP/jM4n+yexFFioOl3F6G+jRJ9rvxzIprgwj5m5aDqMRWILyTVr+Xyj1NXtSIexyJ49xTl5BDhz5gxXrlzJqm/GQALdAcpPlKNYFHZ/sJuBXwZyNiiLfKuJaDwhRevp0xAlJbZ0+/r1IS5fvkdXV1/O2tXV1ZhMJnp6ejL2XT0QNXW5mg3O4vvGRzwcJ3AtkLNBWczHEqm5QwKlpYWMjYWXrItGE7S335Ci39zcTGdnZ8Z+qwYycmcES4EF21upM2ebextTD6aITEWkmMyVmVCUYnseBa/c+Vkta59D6up24PON8vjxcwCOHn2X5uYKKT4BGhoaCIVC9Pb2rtpv1Uk90B2g4lQFNW016XWzwVke/vyQqhZ5D2xaiS0kuf/HMxrcJQxNhEgkVJzFm3g6NcfwZDizwD/Yvr2Aixff5+DB72htraS01I7fP4nLtVma35aWFjo6OnC5XCv2WfVUKnqniH0fLn12qDxdibXQmpOxhalp4qMTOWksEpycw3t/jPlYAlVVCTyZXnMYi5w9W8nNm5+wa1cR4XCc48f3cPv2Kc3ehBA0NTWl24cPH6a1tXXFZxDIUCE1n9b8a53T7cTpdmo2CZD481lO418lPL9AeH5OipbLtVlaVSiKQltbW7othFjSXnaMlC0bSMMIRGdoe0mzAr07UVUVVPhKpq4sBHwpBFQE5e63TNbl9ftrQcUqBA5VxS4EDsAGOFSwC3AIsOn8Uw3wJgSSzYEWOBBYUEFRABUQqYXytw4iVRf6eIhaAV0HIgSfZ3WgU78iQjAMjCdfLgUMozIODGNi3BJmuHyK0MbvSfZIn0MAot13pOjlnTgAgICHwHiSN/dAZ4uuK2QRT5Dlv2z9BzFue3WGEYjOMALRGUYgOsMIRGcYgeiMDb/t9fv99PWlvlM3NjZis9kyjPh/seGBKIqC2ZzarBC6fcf32tjwQDweDx6PZ6M3+8ZgzCE6wwhEZxiB6Ix1mUMW39IarB2jQgwMDAwMDAwM1oO/ADU9gboBWnVoAAAAAElFTkSuQmCC"`;

exports[`Correctly render sequence variation: TC/- without labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAoCAYAAAAIeF9DAAAABmJLR0QA/wD/AP+gvaeTAAABPklEQVRoge3YsS4EURiG4fffFRqFUqOgUGATYlyIQu8KNFsoFLKNzg1ouANRq0VjKxcgTCEuQRRHschOxsSSnPUl8z3VzpndyZ+8OZnNif5LPzFmvzglVVaaRcDG02FlbTjcq32vKIqY7Ik203RjdnGzcn2zXL3eub3IM1HLdf57AKtyEDEOIsZBxDiIGAcR4yBiHESMg4hxEDEOIsZBxDiImNqx+P0SKSVIMPjhh8cR0Cvrz7C/azx+b5SYJ9jKMIsxYZAY7aQVgu2A1QTdj33xkHO4Npp0hxwELACk4C3gKuBsveQ642yt9O07BOD18u5rbW63+Px41O1wvvbI81Sma6FfvUN6JSe5BrER/+0V4yBiHESMg4hxEDEOIsZBxDiIGAcR4yBiHESMg4hpPFwcO+G1KfIOMTMzM8vhHflIJ6BuuNNvAAAAAElFTkSuQmCC"`;

exports[`Correctly render sequence variation: bumped variants with labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAA7CAYAAACNOi92AAAABmJLR0QA/wD/AP+gvaeTAAAGpElEQVR4nO2bW0wc1xnHf2d29gJ74WIvjktpjEGOhYJM18WSXUdOlSpS4rZSpKiK0zwYG1uW0peGyg9RpahvfnDSh8gvsUgsu1GrKJHTvtBYciQrspC2Dr4Q5K1lGwWM8QIxsMsuuzu7M3nAbCDlOnuAWZjfy3KGMx9/zn++M+fMtyPaom0Gkvhk6F0MadGg4yBS460EQkD64jVp8VRpkWbQcrlJUqQbAPznZTnxvquYivenql9KiTf0t6tS4sxEkR7RpiCWlCHxh3Fuf3ab0LEQqsdkUgmB59ln5vxV+n/3MTIZc3EBp6pQ4lZRhCCVyZLK5EzHevw4RV9fjMnJLMFgCXV1FQhhLpau61y4cIHm5mYaGhoA6OnpQdM0mprmzvolZUjPP3voPN3J/Uv3zSmbFpicRE9O4gj4UDdX5tuF3CgCpS5+GvShKgIw2BTwECh1mYp1/XqUc+e6GR9Po6qCmzeH+PTTiGlthmFw5swZTp06lT/W1dVFOBye95wlXe6RzyM895fniFyMsON3O8yqI3PvWwCEqqKUluTbZnEogmB5CX3ROFpOB2BsIoNDWf4lnUhofPFFLydO/JzycjcAzc1bGR1NFaTR6/Xi9/vp7Oxk7969i/ZfNEOGe4ZxljppOtzE4PVB0uPpggTKxONykMpk82ZMk9OXn3EDAxNUV/vzZkxTUeEpSCPA0aNHOXv27JL6LmpI5GKEna/sRHEqbP/1du523C1YoCxUh2Jq8OciHk/j8znz7d7ecb7++hE3bw4VHDsUCuFwOOjq6lq078KGGFPTVaw/Rvj9MFpCI/K5+TlVNtmcbmp6mgu/383EhDbrWC5ncOlSr5T4LS0ttLe3L9pvQUMeXnuIs9SJd4sXgM0Nmxm5PUJyJClFZKGkMjk8LhWnOvvfMGNSdbWPgYE4Y2NTU3JtbRlNTVVSdALs27ePeDxOd3f3gv0WvKlHLkZofKOR3cd354/F+mPc+fcdmo7I2vyZJ6cbDI9NUhP0MZZIY+hQ6lGJJzPEktriAWbg9Tp58cVaPvzwFqHQFvx+F9FoksrKEml6jxw5wsmTJ6mvr5+3z4KGlP2sjGd+O3vvsOvwLoZ7hgsSlh0ZRTjjBcWYJpbMMJnJPtmHwMh4irRmbh8SCm1h27Yy+vpipFJZduyo4KWXtpvWJoTg0KFD+faBAwdobW2ddw8Cixiy+8Tu/zsWbAgSbAiaFgmQ++5xQef/GC2ro2XNbyxnUlnpobKy8JUVgKIoHD9+PN8WQsxqz3mOlL9sIw3bEIshZ834hJpuDMMADP4qI17Hb3jHMOCr/VyREa/vaQ4AvP6xHH0C3hECGvvljeOKPH63Op+9yh+dGgl3iphbY8KTJOZJkfBOEPMlmCgfJVaSQs5NaZlsKEOEgW4IlLSbTWk3myZ88/dVdDSnRmwh434yKF/jhjKkpo+vcgqOnBN3VsGVVXHrDpw5FXdO4MqpuHUFl+7ArSs4FzPuDx/L1yj9HgLw+6o2aTFPV8m9Zv48lJUW65OhdwHob5Q3jvYqy2LYhliMNTEk/jBO+P0w2VTh08fdu6N89FE37733Xzo67pPLFf44Xqa+5bImhsgqCbe33+L55/9Bf38Mt9vB+fPf8Npr/7KMPjOsySpLRkl4aCjJW299yY0bLdTWlgHw5psh7t0bs4Q+s6x6hsgqCYfDg+zZszVvxjR1deWW0GeWVTdEVkl4YCDOU0958+3Ll7/lgw9ucP78N5bQZ5bVNURiSbi62s+jR4lZx9LpHG1tX1pCn1lW1RCZJeE9e7YSDg/S2zsOwAsvPE1LS6Nl9JllVW/qMkvCVVWlnD79K/bv/zutrbuorvZx69Yw9fUVltBnllXNkPlKwi6/uW8aHju2iytXXmfbtjISCY2DB+u4evUNy+gzg/0sqwDsZ1kbgBXJkI2GnSHrGKkZIpvumqkavYGcGviyMHAJQcAw8AlBAPACAQN8AgICvAZssmvqhbKUgRYEEDgxQFEAAxBTH8oPcaZf5JG6SdlQhgjB20sa6KmfkkLwAIjqTz4FPMAgCjzAQdSZ4MHOEeR8BXNao8xgsumumVoknLsmZxl9+BdTy1QBd4CozuoN9FLZUBkyzbP9zP2yowWwV1kWY10YspYlV9msC0PWsuQqm3VhyMySa7FT9IasdclVNkVvyFqXXGVT3IZYoOQqm6I2xAolV9kU9cbQCiVX2RR1hlih5Cqbos6QlXpLeC0p6gxZjxTF017ZyCwoycbOEBsbGxsbGxubleB7nxP+UDtlBh8AAAAASUVORK5CYII="`;

exports[`Correctly render sequence variation: bumped variants without labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAA7CAYAAACNOi92AAAABmJLR0QA/wD/AP+gvaeTAAACDklEQVR4nO3ZvWtTYRiG8ettBEUyNAoZlOLHIKgEFDP5QaWb1tnBSf8BQeLmUFycDO4uurh0EacI4lCKBaGD0MlJayhiwI9BpKLldYgxPaTRePKKN5z7NzVPk/DA1dOE84bGu0Zkk/lOk5iZQGuWgdkwIcDXh8uZWb1eD6O92rYN+8WVp8c2PXoBwOPz/dn7Snd2rXrq16xz51ni9Ypn4n8vYFkOIsZBxDiIGAcR4yBiHESMg4hxEDEOIsZBxDiIGAcRM3BbfGqFGCMQudmbtS4wFyMsnmahN3uzj2mASw/6zwswFwLU2oPva6MZ+wpZOsmZT5OUUyxjvzkP+ZPSBusbJXa8PsDM6n7OVj7y8lwr5WrFlDvI3jWef9lJ5XOZPevb2f1hF4dTLlZUW36GAFysNjLz29XBdtc73zOP5ztNANo1f4bk5W9ZYhxEjIOIcRAxDiLGQcQ4iBgHEeMgYhxEjIOIcRAxDiJm6N3ecfhub36+QsSM9Je8MtU9Z4/0z88zImUCxwPM+Ex9PLlPDEM35kECJwIcilD6meFVquWKKHcQ4GqASYAY+BbgUYC7R9s8SbRbIY38Lwvg/nL/WPdyvdn78UZpgntHVnmbfLsCGucKAaDW5laKRazL37LEOIgYBxHjIGIcRIyDiHEQMQ4ixkHEOIgYBxHjIGL+6m7vVnwYlZavEDMzM7N/4QeOWV3sfEOdUAAAAABJRU5ErkJggg=="`;

exports[`Correctly render sequence: ATCGN with labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAOCAYAAADZjbloAAAABmJLR0QA/wD/AP+gvaeTAAADBklEQVRYhe2XX0gTcQDHP7c8maWOucgwtFy4gUXQywYFRVoYFqhQYQxs5R+IHsKEeuhBiCLqqZ4WZUH4YEnhpmEh2UN/kCzD/rEOMyqZNtFtLduVtl0PFhTEprsLgvZ5Oo7vfe97v+/d734/ocnfpKAhB2oDWtpxumSPpn6uxpua+g0M7NTUT6epWwrVpM1FNNQ9ROfeTpz3nJgspqRuJIhp5F84NXus18MCHcrnCACjh0/ybXwiKd9FepEdG1dizstGJwj4gxGu3x1mbDIyb6/JSZlDh+7Q1+djZiZGbu4ijhyxU1VlSSpbNBrFbrfjcDhobGwEoLW1FVmWaWho+OM1cypEcksssy9DckusO7wuqXDKzDfeO5sAyK7YQpoph8Clq0l5/UQA9les4rXvI8cuPyIaUzDnZbPYkDHvQhQFtm+/xqZNBbx8WYco6pCkAFeueJMuBECv19Pb24vT6cRoNCbUJ5yypqem8T30UXamjFcdr5IO9jdYvjSL7IXpdD54SzQ2+yt8Mxrm+ZvJeXv1948yMSFz/PgGRHF2WKzWHJqb16vKKIoilZWVtLW1zUmfsJDhW8MUlhRiNBvJMGXgf+pXFVBLluYsZGR8ipiifl0iSQHWrs1FpxM0SPY71dXVdHd3Ew6HE2oTFuLt8GKttAJgrbD+c18JGo6f8kuxBw/exmw+R1HRedW+mZmZlJeX097enlAbtxA5KDNyf4Seph5abC0MuAaQPBJKTNOVctJ8CEQoWJKFTlDfitWaw+DgOLEfz3b27GaePdtHKPRFtTeAw+HA7XYjy3JcXdxChrqGKN5VTP3jeur666h/Uo9hhQHfQ58mIdXy7sMnwpFpKtYXkrZgtpSVeQbWmOe/ErTZ8jCZMjh69C5fv0YBCAa1KQPAYDBQWlqKx+OJq4tbiLfDS1F50W/nLNss/8y0pQAuzwsMmek077FxotbOVls+/mD8t/BPCALcuLGDsbEpVq++iMVynt27u3C5yjTLW1NTQygUip8jtVNXR2qnniJFiv+Y73i6DXKFiVwRAAAAAElFTkSuQmCC"`;

exports[`Correctly render sequence: ATCGN with lower case labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAOCAYAAADZjbloAAAABmJLR0QA/wD/AP+gvaeTAAAClklEQVRYhe2XT0iTYRzHP/vji3/aXDQ1ZsOh2WHSyEiSUZfAhYlIQUMP4iHyUJCGXTt1CJpXCQUlhEBDSTHsVOwQgtVwSEqQyC6+y2w6xW2vvlt2kKgOm/K+LwNxn/Pv/fx+PF+e53kfXc9qzx4acv/OupY6nl3r0NT3/OFbTX2BwG1NfXpNbTlUk/VAdIKAqeFqttseGbIeiD5fwNziyXbbI4PxoIIx7xjShkRyJ0ndvTpqWmtUNbR4mzGWWTn95BE7X5fYeDmhygdw8VwJNy5XAPAjmmBgakGxq7f3I4OD89jtJhyOYpxOK93dlxS5IpEInZ2duN1uwuEwsizj8/kQBCHtNwcG0tTfRMHJAuSEzEjzCNVN1Qim9MKDiL6aIv+Ck++PexU7/qXEUsDNK5X4RufYiu1SlJ+n2LW4+JPh4S8EAh3odDrq64dxOq2q5hNFkba2Nmw2Gz6fD7/fj8eT/oTIHMgeBIeChN6HMOYb2Q5vEw1FKT1fqmpILTlbXsz8coSt2C4AMUlW7JqZWaGxsZLCwv1QW1qqVc9nt9ux2WwAOBwORFHMWJ8xkJA/hPhZxPvai0EwMN46Tmo3pXrI44TR+HeJ9Xo9qVTm9ct4qUubEpYKCwbBQHwtjvgpc7qH4VdCwmAxq/b8YWllE1flKYqL9o/REwXKjyy3u5zp6WXicZlEIsnk5Detxjw0GXdI1fUqFkYXmLo7hVAkUOYqU91wT04Sn53jTP9T4rNB1odGVfnWogkmPizz4JYLvV5HOBJj4M2iIpfTaaW9vYba2hfY7SZqa8swm5Xfl0rQ5V7q/xOPyxQW5iFJSRoaRunr8+BylaSt1/qlfuBf1nGjq+sdweAqkpSivb0mYxg5cuTINr8BuyLSah7LHrkAAAAASUVORK5CYII="`;

exports[`Correctly render sequence: ATCGN without labels 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAOCAYAAADZjbloAAAABmJLR0QA/wD/AP+gvaeTAAAAZklEQVRYhe3RoRGAMBBE0YSargJ6QCOQUUy6oAFKoYoUgqEFornDs8z8J1ftzM/1rHdyynL5KW3jHLZ9PcLW2hQ2M8thxKvh6wN4IogYgoghiBiCiCGIGIKIIYgYgoghiBiCAPiRDvAiChi7/b8vAAAAAElFTkSuQmCC"`;
